### PR TITLE
Get ca and server name

### PR DESCRIPTION
**Pull request by quoctri1**
@HuuPhuc25 please review my change.

### CODE REVIEW

File: applications/lib_app/rsa_util/rsa_util.cpp [hunk 1]
```diff
@@ -252,3 +252,42 @@ bool openssl_gen_dh(const char* dh_key, int bitsize)
...
+bool get_server_name(const char *server_crt_path, char *server_name)
+{
+    char temp[256];
+
+    FILE *f = fopen(server_crt_path, "r");
+    if (f == NULL) {
+        return false;
+    }
+
+    while (fgets(temp, sizeof(temp), f) > 0) {
```
Comment (HuuPhuc25) [inline applications/lib_app/rsa_util/rsa_util.cpp]: You can use a command line to get subject: openssl x509 -noout -subject -in server.crt

File: applications/lib_app/rsa_util/rsa_util.cpp [hunk 2]
```diff
@@ -252,3 +252,42 @@ bool openssl_gen_dh(const char* dh_key, int bitsize)
...
+
+    snprintf(command_line, sizeof(command_line), "openssl x509 -in %s -noout -text", ca_crt_path);
+
+    FILE *f = popen(command_line, "r");
+    if (f == NULL) {
+        return false;
+    }
+
+    while (fgets(temp, sizeof(temp), f) > 0) {
+        sscanf(temp, "\t\tSubject: %[^\n]", ca_name);
```
Comment (hhdang6637) [inline applications/lib_app/rsa_util/rsa_util.cpp]: What happen if ca_name isn't long enough ?

Comment (HuuPhuc25) [reply applications/lib_app/rsa_util/rsa_util.cpp]: Tri should use a command to get all subject of ca: openssl x509 -noout -subject -in ca.crt. There are result for this command: subject= /CN=CA/ST=NC/C=VN/emailAddress=rootCA@gmail.com/O=Example/OU=IT Department.

File: applications/lib_app/rsa_util/rsa_util.cpp [hunk 3]
```diff
@@ -252,3 +252,42 @@ bool openssl_gen_dh(const char* dh_key, int bitsize)
...
+    while (fgets(temp, sizeof(temp), f) > 0) {
+        sscanf(temp, "\t\tSubject: %[^\n]", server_name);
+    }
+
+    fclose(f);
+
+    return true;
+}
+
+bool get_ca_name(const char *ca_crt_path, char *ca_name)
```
Comment (HuuPhuc25) [inline applications/lib_app/rsa_util/rsa_util.cpp]: the name of a method is get_ca_subject, it's ok?

**Pull request closed without merging**